sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -185,13 +185,13 @@ public void write(int ordinal, Decimal input, int precision, int scale) {
       // grow the global buffer before writing data.
       holder.grow(16);
 
+      // zero-out the bytes
```

Comment:
nit: Can we refine a comment like the following to avoid this problem in the future?
```
// always zero-out the 16-byte buffer
```

Reply:
+1